app/controllers/admin/facility_groups_controller.rb

```diff
@@ -60,7 +51,10 @@ def set_protocols
 
   def set_facility_group
     @facility_group = FacilityGroup.friendly.find(params[:id])
```

Comment:
Here is a good example of the danger of doing first class model look ups - someone could craft a URL to directly hit the `destroy` action and blow away any FacilityGroup here. Maybe Pundit would catch this for us via its automatic filtering?  In any case, this could be written like this:

```
current_admin.facility_groups.friendly.find(params[:id])
```

Reply:
I feel like this comment actually makes the point **for** Pundit in some ways, rather than against it. If it makes us squirm to look top-level model calls because who-knows-what-might-happen type scenarios, then we should probably invest in something that handles this for us automatically.

Pundit doesn't allow you to render a controller action if you don't authorize it, it'll fail in development, it'll fail the specs and it'll fail if you launch it in production. It'll behave the same way for the `destroy` action as well.

Reply:
> If it makes us squirm to look top-level model calls because who-knows-what-might-happen type scenarios, then we should probably invest in something that handles this for us automatically.

Yes, exactly! Thats why we are adding a concept of 'role' or 'access' to our domain model, and then we can rely on the database and ActiveRecord to "handle this for us automatically" 😀😀😀 .  

We have the tools in our framework to handle many of these cases, its just SQL and good data modeling and then using the ORM we have.  No need for another auth-specific framework if our use case is right in the sweet spot of literally what Rails and ActiveRecord were built for.